src/Symfony/Component/Validator/Constraints/MacAddressValidator.php

```diff
@@ -23,6 +23,71 @@
  */
 class MacAddressValidator extends ConstraintValidator
 {
+    /**
+     * Checks whether an MAC address is valid.
+     */
+    private static function checkMac(string $mac, mixed $type): bool
+    {
+        if (!filter_var($mac, \FILTER_VALIDATE_MAC)) {
+            return false;
+        }
+
+        return match ($type) {
+            MacAddress::LOCAL_ALL => self::isLocal($mac),
+            MacAddress::LOCAL_NO_BROADCAST => self::isLocal($mac) && !self::isBroadcast($mac),
+            MacAddress::LOCAL_UNICAST => self::isLocal($mac) && self::isUnicast($mac),
+            MacAddress::LOCAL_MULTICAST => self::isLocal($mac) && !self::isUnicast($mac),
+            MacAddress::LOCAL_MULTICAST_NO_BROADCAST => self::isLocal($mac) && !self::isUnicast($mac) && !self::isBroadcast($mac),
+            MacAddress::UNIVERSAL_ALL => !self::isLocal($mac),
+            MacAddress::UNIVERSAL_UNICAST => !self::isLocal($mac) && self::isUnicast($mac),
+            MacAddress::UNIVERSAL_MULTICAST => !self::isLocal($mac) && !self::isUnicast($mac),
+            MacAddress::UNICAST_ALL => self::isUnicast($mac),
+            MacAddress::MULTICAST_ALL => !self::isUnicast($mac),
+            MacAddress::MULTICAST_NO_BROADCAST => !self::isUnicast($mac) && !self::isBroadcast($mac),
+            MacAddress::BROADCAST => self::isBroadcast($mac),
+            MacAddress::ALL_NO_BROADCAST => !self::isBroadcast($mac),
+            default => true,
+        };
```

Comment:
shouldn't we use a bitfield instead of this match?

Reply:
You mean using bit-operators instead of checking hexadecimal chars? Did not want to convert mac address to hex (sanitize) and then to binary/int, just for validation. I decided to solve it in a validator as simple as possible. For some more specific tasks (convertion [int, bin, hex, string], generator, next/prev, ...) I wrote an own DependencyInjection, but for simplification this is also relying on hex or int (e.g. for broadcast validation). For me it's easier to read/understand/debug/maintain and performance is also not a big deal here (the conversion we need before using bit operators also needs some resources).
I have another DependencyInjection for CIDR, here I used bit-operators, because IpV6 is a bit bigger than int max and you have nice possibilities to split the blocks (netmask...) by flipping some bits. But it's hard to understand and so to maintain.
But if u prefer using bit operators, I can try implementing it as compact as possible, but I cannot see much advantages here, just that it's rly nerdy/cool (I also like!) :D

Reply:
Oh I think I misunderstood. U mean using bit/int-flags like the options-parameter of the `filter_var` function (https://www.php.net/manual/en/function.filter-var.php)?

Let me think about it, may I'll find a nice solution the next days :)

Reply:
ok, maybe we can keep it as is :shrug: